### PR TITLE
Refactor/execute error

### PR DESCRIPTION
**Pull request by themanforfree**
#160

### CODE REVIEW

**Comment by themanforfree** [comment]
based on #164

File: xline/src/server/lock_server.rs [hunk 1]
```diff
@@ -36,7 +36,6 @@ const DEFAULT_SESSION_TTL: i64 = 60;
 
 /// Lock Server
 //#[derive(Debug)]
```
Comment (Phoenix500526) [inline xline/src/server/lock_server.rs:38]: Plz remove this line.

File: xline/src/storage/execute_error.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    pub(crate) fn lease_already_exists(lease_id: i64) -> Self {
+        Self::LeaseError(format!("lease {} already exists", lease_id))
+    }
+
+    /// Auth is not enabled
+    pub(crate) fn auth_not_enabled() -> Self {
+        Self::AuthError("auth is not enabled".to_owned())
+    }
+
+    /// Auth Failed
```
Comment (Phoenix500526) [inline xline/src/storage/execute_error.rs:61]: Auth failed. Plz keep the comment style consistent.

File: xline/src/storage/db.rs [hunk 3]
```diff
@@ -117,13 +90,10 @@ where
...
                 let _prev_val = storage.insert(key, value).map_err(|e| {
-                    ExecuteError::InvalidCommand(format!(
-                        "Failed to insert revision {:?}: {}",
-                        new_rev, e
-                    ))
-                });
-                Ok(del_kv)
-            })
-            .collect()
+                    ExecuteError::db_error(format!("Failed to insert key-value, error: {e}"))
```
Comment (Phoenix500526) [inline xline/src/storage/db.rs:93]: It looks that there is nothing different between `ExecuteError::db_error(format!("Failed to insert key-value, error: {e}"))` and `ExecuteError::DbError(format!("Failed to insert key-value, error: {e}"))`. So why do need such a function?

Comment (themanforfree) [reply xline/src/storage/db.rs]: removed

File: xline/src/storage/execute_error.rs (hunk 2 above)
Comment (themanforfree) [reply xline/src/storage/execute_error.rs]: fixed

File: xline/src/server/lock_server.rs (hunk 1 above)
Comment (themanforfree) [reply xline/src/server/lock_server.rs]: fixed

**Review by Phoenix500526 — APPROVED** [review]

File: curp/src/test_utils/execute_error.rs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    #[error("meet io related error")]
+    IoError(String),
+}
+
+impl From<io::Error> for ExecuteError {
+    #[inline]
+    fn from(err: io::Error) -> Self {
+        Self::IoError(err.to_string())
+    }
+}
```
Comment (markcty) [inline curp/src/test_utils/execute_error.rs:23]: I guess there is no need to have such a complex execute error for test command. String would be fine.

File: curp/tests/common/mod.rs [hunk 5]
```diff
@@ -22,3 +27,23 @@ pub async fn sleep_millis(n: u64) {
...
 }
+
+/// Error met when executing commands
+#[allow(clippy::module_name_repetitions)] // this-error generate code false-positive
+#[non_exhaustive]
+#[derive(Error, Debug, Clone)]
+pub enum ExecuteError {
+    /// Command is invalid
+    #[error("invalid command {0} ")]
+    InvalidCommand(String),
```
Comment (markcty) [inline curp/tests/common/mod.rs:38]: same above

File: xline/src/storage/lease_store/mod.rs [hunk 6]
```diff
@@ -277,11 +277,7 @@ impl LeaseStore {
 
     /// Keep alive a lease
     pub(crate) fn keep_alive(&self, lease_id: i64) -> Result<i64, ExecuteError> {
-        if !self.is_leader() {
-            return Err(ExecuteError::InvalidCommand(
-                "lease keep alive must be called on leader".to_owned(),
-            ));
-        }
+        assert!(self.is_leader(), "keep_alive must be called on leader");
```
Comment (markcty) [inline xline/src/storage/lease_store/mod.rs:280]: I guess it leadership change can happen before this function is called. If so, assert! will panic. So I think the previous solution is better.

**Review by markcty — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**